tests/distributed/test_mp_dataloader.py

```diff
@@ -0,0 +1,123 @@
+import dgl
+import unittest
+import os
+from dgl.data import CitationGraphDataset
+from dgl.distributed import sample_neighbors
+from dgl.distributed import partition_graph, load_partition, load_partition_book
+import sys
+import multiprocessing as mp
+import numpy as np
+import time
+from utils import get_local_usable_addr
+from pathlib import Path
+from dgl.distributed import DistGraphServer, DistGraph, DistDataLoader
+import pytest
+
+
+class NeighborSampler(object):
+    def __init__(self, g, fanouts, sample_neighbors):
+        self.g = g
+        self.fanouts = fanouts
+        self.sample_neighbors = sample_neighbors
+
+    def sample_blocks(self, seeds):
+        import torch as th
+        seeds = th.LongTensor(np.asarray(seeds))
+        blocks = []
+        for fanout in self.fanouts:
+            # For each seed node, sample ``fanout`` neighbors.
+            frontier = self.sample_neighbors(
+                self.g, seeds, fanout, replace=True)
+            # Then we compact the frontier into a bipartite graph for message passing.
+            block = dgl.to_block(frontier, seeds)
+            # Obtain the seed nodes for next layer.
+            seeds = block.srcdata[dgl.NID]
+
+            blocks.insert(0, block)
+        return blocks
+
+
+def start_server(rank, tmpdir, disable_shared_mem, num_clients):
+    import dgl
+    print('server: #clients=' + str(num_clients))
+    g = DistGraphServer(rank, "mp_ip_config.txt", num_clients,
+                        tmpdir / 'test_sampling.json', disable_shared_mem=disable_shared_mem)
+    g.start()
+
+
+def start_client(rank, tmpdir, disable_shared_mem, num_workers):
+    import dgl
+    import torch as th
+    os.environ['DGL_DIST_MODE'] = 'distributed'
+    dgl.distributed.rpc.reset()
+    dgl.distributed.init_rpc("mp_ip_config.txt", num_workers=4)
+    gpb = None
+    if disable_shared_mem:
+        _, _, _, gpb, _ = load_partition(tmpdir / 'test_sampling.json', rank)
+    train_nid = th.arange(202)
+    dist_graph = DistGraph("mp_ip_config.txt", "test_mp", gpb=gpb)
+
+    # Create sampler
+    sampler = NeighborSampler(dist_graph, [5, 10],
+                              dgl.distributed.sample_neighbors)
+
+    # Create PyTorch DataLoader for constructing blocks
+    dataloader = DistDataLoader(
+        dataset=train_nid.numpy(),
+        batch_size=32,
+        collate_fn=sampler.sample_blocks,
+        shuffle=True,
+        drop_last=False,
+        num_workers=4)
+
+    dist_graph._init()
+    for epoch in range(3):
+        for idx, blocks in enumerate(dataloader):
+            print(blocks)
+            print(blocks[1].edges())
+            print(idx)
```

Comment:
can we check the sampled subgraphs instead of printing the edges? it'll be a lot of printed messages. also printing doesn't capture any errors in CI.